packages/next/src/server/base-server.ts

```diff
@@ -1776,25 +1776,27 @@ export default abstract class Server<
     )
   }
 
-  private getWaitUntil(): WaitUntil | undefined {
+  protected getWaitUntil(): WaitUntil | undefined {
     const builtinRequestContext = getBuiltinRequestContext()
     if (builtinRequestContext) {
       // the platform provided a request context.
       // use the `waitUntil` from there, whether actually present or not --
       // if not present, `unstable_after` will error.
-      return builtinRequestContext.waitUntil
-    }
 
-    if (process.env.__NEXT_TEST_MODE) {
```

Comment:
this was hiding the issue in our tests. it's a leftover that should've been removed but wasn't.